source/Energinet.Charges.Libraries/source/Energinet.DataHub.Charges.Clients.SimpleInjector/ContainerExtensions.cs

```diff
@@ -48,8 +48,32 @@ public static class ContainerExtensions
                 new ServiceBusRequestSenderProvider(serviceBusClient, serviceBusRequestSenderConfiguration));
 
             container.Register<IDefaultChargeLinkClient, DefaultChargeLinkClient>(Lifestyle.Scoped);
+        }
+
+        /// <summary>
+        /// SimpleInjector extension for registering Energinet.DataHub.Charges.Clients NuGet package.
+        /// </summary>
+        /// <param name="container"></param>
+        /// <param name="serviceBusClientCreator">ServiceBusClient creator func required to connect to the shared service bus namespace</param>
+        /// <param name="serviceBusRequestSenderConfiguration"></param>
+        public static void AddDefaultChargeLinkClient(
+            this Container container,
+            [DisallowNull] Func<ServiceBusClient> serviceBusClientCreator,
+            [DisallowNull] IServiceBusRequestSenderConfiguration serviceBusRequestSenderConfiguration)
+        {
+            if (container == null)
```

Comment:
I think you should replace the whole body of the method with something like:
```suggestion
            if (serviceBusClientCreator == null)
                throw new ArgumentNullException(nameof(serviceBusClientCreator));

            container.AddDefaultChargeLinkClient(serviceBusClientCreator.Invoke(), serviceBusRequestSenderConfiguration();
```